template/util.template

```diff
@@ -35,10 +33,14 @@ module.exports = {
     }
 
     if (node.children) {
-      node.children.forEach(getDataStructure)
+      node.children.forEach(processNode)
     }
 
-    return [subjects, aspects, samples];
+    node.samples = samples;
+    node.subjects = subjects;
+    node.aspects = aspects;
```

Comment:
This will add samples/subjects/aspects fields at each level of the hierarchy, which won't necessarilly correspond to the samples/subjects/aspects under that node. Seems confusing - why not return [subjects, aspects, samples] as before?